addons/addon-base-raas/packages/base-raas-services/lib/storage-gateway/__tests__/storage-gateway-service.test.js

```diff
@@ -498,11 +498,80 @@ describe('storageGateway', () => {
           rev: 0,
           createdBy: userId,
           updatedBy: userId,
-          createdAt: expect.any(String),
-          updatedAt: expect.any(String),
         }),
       );
       expect(dbService.table.update).toHaveBeenCalled();
     });
   });
+
+  describe('createFileShare', () => {
+    it('should return existing file share', async () => {
+      // BUILD
+      dbService.table.get.mockImplementationOnce(() => {
+        return { file_share_s3_locations: { s3LocationArn: 'fileShareArn' } };
+      });
+      // OPERATE
+      const result = await storageGateway.createFileShare(context, 'gatewayArn', 's3LocationArn', 'roleArn');
+      // CHECK
+      expect(result).toEqual('fileShareArn');
+    });
+
+    it('should throw error if SG record can not be found', async () => {
+      // OPERATE and CHECK
+      try {
+        await storageGateway.createFileShare(context, 'gatewayArn', 's3LocationArn', 'roleArn');
+        expect.hasAssertions();
+      } catch (err) {
+        expect(storageGateway.boom.is(err, 'notFound')).toBe(true);
+        expect(err.message).toContain('storage gateway with id "gatewayArn" does not exist');
+      }
+    });
+
+    it('should create new file share and save to DDB', async () => {
+      // BUILD
+      dbService.table.item.mockClear();
+      dbService.table.get
+        .mockReturnValueOnce({
+          file_share_s3_locations: { anotherS3LocationArn: 'anotherFileShareArn' },
+        })
+        .mockReturnValueOnce({
+          elasticIP: '10.52.4.93',
+          fileShares: { anotherS3LocationArn: 'anotherFileShareArn' },
+          rev: 3,
+        })
+        .mockReturnValueOnce({
+          elasticIP: '10.52.4.93',
+          fileShares: { anotherS3LocationArn: 'anotherFileShareArn' },
+          rev: 3,
+        });
+
+      const expectedRequest = {
+        ClientToken: expect.any(String),
+        GatewayARN: 'gatewayArn',
+        LocationARN: 's3LocationArn',
+        Role: 'roleArn',
+        ClientList: ['10.52.4.93/32'],
```

Comment:
Probably put something more random if this is an original IP address

Reply:
This was a IP address I made up, so it should be okay.